### PR TITLE
[Doc]OpenPAI deployment document refactor

### PR DESCRIPTION
**Pull request by YanjieGao**
Entry link:
1. deploy: https://github.com/Microsoft/pai/blob/yanjga/doc_deploy/pai-management/doc/cluster-bootup.md
2. configuration:
https://github.com/Microsoft/pai/blob/yanjga/doc_deploy/pai-management/doc/how-to-write-pai-configuration.md

 
Refactor Goal:
1. Detailed steps on each step to deploy a customized cluster. 
2. Checks for each step so that users can ensure things are going right. 
3. Some problem investigation skills and problem solving experiences for deployment/upgrading.

Note:
1. Combine deploy of customize and quick deploy, highlight single box deploy. Make quick deploy as first step to let user use minimal config to get the correct config file. Then user could customize the config based on requirements.
2. Make configuration as two view and index: One is user could config from specific scenarios like config virtual cluster. Other is config from each file.
3. Add troubleshooting part. 
4 Index Maintenance part.
5 Reformat configuration properties.

### CODE REVIEW

File: pai-management/doc/cluster-bootup.md [hunk 1]
```diff
@@ -8,32 +8,57 @@ Please refer to Section [single box deployment](./single-box-deployment.md) if u
...
+
+### Step 0. Prepare the dev-box <a name="c-step-0"></a>
 
-## Quick deploy with default settings <a name="quickdeploy"></a>
-### Step 0. Prepare the dev-box
 It is recommended to perform the operations below in a dev box.
 Please refer to this [section](./how-to-setup-dev-box.md) for the details of setting up a dev-box.
 
-### Step 1. Prepare the quick-start.yaml file <a name="step-1a"></a>
+#### How to check
```
Comment (hao1939) [inline pai-management/doc/cluster-bootup.md]: Hi @YanjieGao , we may need to add some instructions about starting a dev-box.

File: pai-management/doc/cluster-bootup.md [hunk 2]
```diff
@@ -62,15 +87,17 @@ ssh-password: pai-password
...
+check all configruation items of the quick-start.yaml are correct.
 
-### Step 2. Generate OpenPAI configuration files
+### Step 2. Generate OpenPAI configuration files <a name="c-step-2"></a>
 
 After the quick-start.yaml is ready, use it to generate four configuration yaml files as follows.
 
 ```
 python paictl.py cluster generate-configuration -i ~/quick-start.yaml -o ~/pai-config -f
```
Comment (hao1939) [inline pai-management/doc/cluster-bootup.md]: it should executed under the /pai/pai-management directory in the dev-box.

File: pai-management/doc/cluster-bootup.md [hunk 3]
```diff
@@ -62,15 +87,17 @@ ssh-password: pai-password
...
 ```
-
+#### How to check
 The command will generate the following four yaml files.
```
Comment (hao1939) [inline pai-management/doc/cluster-bootup.md]: those files under the directory `-o ~/pai-config`

File: pai-management/doc/cluster-bootup.md (hunk 1 above)
Comment (YanjieGao) [reply pai-management/doc/cluster-bootup.md]: will try to add the main instructions here

File: pai-management/doc/cluster-bootup.md (hunk 3 above)
Comment (YanjieGao) [reply pai-management/doc/cluster-bootup.md]: ok

File: pai-management/doc/cluster-bootup.md [hunk 4]
```diff
@@ -1,41 +1,105 @@
...
+
+### Step 0. Prepare the dev-box <a name="c-step-0"></a>
 
-## Quick deploy with default settings <a name="quickdeploy"></a>
-### Step 0. Prepare the dev-box
 It is recommended to perform the operations below in a dev box.
-Please refer to this [section](./how-to-setup-dev-box.md) for the details of setting up a dev-box.
 
-### Step 1. Prepare the quick-start.yaml file <a name="step-1a"></a>
+```dev-box``` is a docker container used to boot up or/and maintain a PAI cluster. For convenience, we provide a prebuild Docker image on Docker Hub.
```
Comment (sterowang) [inline pai-management/doc/cluster-bootup.md]: >d [](start = 3, length = 1)

Capitalize the first letter?

**Comment by YanjieGao** [comment]
Hi @ydye , this pr build meet jenkins script failed.
seems lack request at   File "/pai/pai-management/k8sPaiLibrary/monitorTool/check_pod_ready_status.py", line 24, in <module>

image

jenkins:
http://40.76.53.138:33080/blue/organizations/jenkins/pai/detail/PR-1025/19/pipeline

**Pull request merged**